--- a/src/Makefile.am.features	2018-08-16 19:04:11.000000000 -0700
+++ b/src/Makefile.am.features	2019-10-15 06:53:07.925635549 -0700
@@ -310,16 +310,6 @@
 all_cairo_private += $(cairo_glesv3_private)
 all_cairo_cxx_sources += $(cairo_glesv3_cxx_sources)
 all_cairo_sources += $(cairo_glesv3_sources)
-if CAIRO_HAS_GLESV3_SURFACE
-enabled_cairo_headers += $(cairo_glesv3_headers)
-enabled_cairo_private += $(cairo_glesv3_private)
-enabled_cairo_cxx_sources += $(cairo_glesv3_cxx_sources)
-enabled_cairo_sources += $(cairo_glesv3_sources)
-endif
-all_cairo_pkgconf += cairo-glesv3.pc
-if CAIRO_HAS_GLESV3_SURFACE
-enabled_cairo_pkgconf += cairo-glesv3.pc
-endif
 
 unsupported_cairo_headers += $(cairo_cogl_headers)
 all_cairo_headers += $(cairo_cogl_headers)
--- a/boilerplate/Makefile.am.features	2018-08-16 19:04:11.000000000 -0700
+++ b/boilerplate/Makefile.am.features	2019-10-15 06:52:15.378559734 -0700
@@ -236,12 +236,7 @@
 all_cairo_boilerplate_private += $(cairo_boilerplate_glesv3_private)
 all_cairo_boilerplate_cxx_sources += $(cairo_boilerplate_glesv3_cxx_sources)
 all_cairo_boilerplate_sources += $(cairo_boilerplate_glesv3_sources)
-if CAIRO_HAS_GLESV3_SURFACE
-enabled_cairo_boilerplate_headers += $(cairo_boilerplate_glesv3_headers)
-enabled_cairo_boilerplate_private += $(cairo_boilerplate_glesv3_private)
-enabled_cairo_boilerplate_cxx_sources += $(cairo_boilerplate_glesv3_cxx_sources)
-enabled_cairo_boilerplate_sources += $(cairo_boilerplate_glesv3_sources)
-endif
+
 
 unsupported_cairo_boilerplate_headers += $(cairo_boilerplate_cogl_headers)
 all_cairo_boilerplate_headers += $(cairo_boilerplate_cogl_headers)
