Index: git/mkspecs/linux-oe-g++/qmake.conf
===================================================================
--- git.orig/mkspecs/linux-oe-g++/qmake.conf	2017-05-31 11:43:10.348498136 -0700
+++ git/mkspecs/linux-oe-g++/qmake.conf	2017-05-31 11:57:09.695989689 -0700
@@ -39,5 +39,13 @@
 
 include(../oe-device-extra.pri)
 
+QMAKE_LIBS_EGL        += -lEGL
+QMAKE_LIBS_OPENGL_ES2 += -lEGL -lGLESv2
+QMAKE_LIBS_OPENVG     += -lEGL -lOpenVG
+
+QMAKE_CFLAGS                += -DLINUX=1
+QMAKE_CXXFLAGS              += -DLINUX=1
+QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO += -DLINUX=1
+
 load(device_config)
 load(qt_config)
