From 5b9f725261076604283913dff275c59f3fd08e76 Mon Sep 17 00:00:00 2001
From: Gary Bisson <gary.bisson@boundarydevices.com>
Date: Tue, 6 Jun 2023 18:16:33 +0200
Subject: [PATCH 6/8] imx/run.sh: remove bashism

Signed-off-by: Gary Bisson <gary.bisson@boundarydevices.com>
---
 imx/run.sh | 79 ++++++++++++++++++++++++++----------------------------
 1 file changed, 38 insertions(+), 41 deletions(-)

diff --git a/imx/run.sh b/imx/run.sh
index 07c9814b2..18b1a51f2 100755
--- a/imx/run.sh
+++ b/imx/run.sh
@@ -5,7 +5,7 @@
 # (c) Basler 2020
 #
 
-RUN_SCRIPT=`realpath -s $0`
+RUN_SCRIPT=`realpath $0`
 RUN_SCRIPT_PATH=`dirname $RUN_SCRIPT`
 echo "RUN_SCRIPT=$RUN_SCRIPT"
 echo "RUN_SCRIPT_PATH=$RUN_SCRIPT_PATH"
@@ -15,26 +15,26 @@ LOCAL_RUN="0" # search modules in /lib/modules and libraries in /usr/lib
 RUN_OPTION=""
 LOCAL_RUN="0"
 # an array with the modules to load, insertion order
-declare -a MODULES=("imx8-media-dev" "vvcam-dwe" "vvcam-isp" "vvcam-video")
+MODULES="imx8-media-dev vvcam-video vvcam-dwe vvcam-isp"
 
 USAGE="Usage:\n"
-USAGE+="run.sh -c <isp_config> &\n"
-USAGE+="Supported configurations:\n"
-USAGE+="\tbasler_1080p60         - single basler camera on MIPI-CSI1, 1920x1080, 60 fps\n"
-USAGE+="\tdual_basler_1080p60    - dual basler cameras on MIPI-CSI1/2, 1920x1080, 60 fps\n"
-USAGE+="\tbasler_4k              - single basler camera on MIPI-CSI1, 3840x2160, 30 fps\n"
-USAGE+="\tbasler_1080p60hdr      - single basler camera on MIPI-CSI1, 1920x1080, 60 fps, HDR configuration\n"
-USAGE+="\tdual_basler_1080p60hdr - dual basler cameras on MIPI-CSI1/2, 1920x1080, 60 fps, HDR configuration\n"
-USAGE+="\tbasler_4khdr           - single basler camera on MIPI-CSI1, 3840x2160, 30 fps, HDR configuration\n"
+USAGE=$USAGE"run.sh -c <isp_config> &\n"
+USAGE=$USAGE"Supported configurations:\n"
+USAGE=$USAGE"\tbasler_1080p60         - single basler camera on MIPI-CSI1, 1920x1080, 60 fps\n"
+USAGE=$USAGE"\tdual_basler_1080p60    - dual basler cameras on MIPI-CSI1/2, 1920x1080, 60 fps\n"
+USAGE=$USAGE"\tbasler_4k              - single basler camera on MIPI-CSI1, 3840x2160, 30 fps\n"
+USAGE=$USAGE"\tbasler_1080p60hdr      - single basler camera on MIPI-CSI1, 1920x1080, 60 fps, HDR configuration\n"
+USAGE=$USAGE"\tdual_basler_1080p60hdr - dual basler cameras on MIPI-CSI1/2, 1920x1080, 60 fps, HDR configuration\n"
+USAGE=$USAGE"\tbasler_4khdr           - single basler camera on MIPI-CSI1, 3840x2160, 30 fps, HDR configuration\n"
 
-USAGE+="\tos08a20_1080p60         - single os08a20 camera on MIPI-CSI1, 1920x1080, 60 fps\n"
-USAGE+="\tdual_os08a20_1080p60    - dual os08a20 cameras on MIPI-CSI1/2, 1920x1080, 60 fps\n"
-USAGE+="\tos08a20_4k              - single os08a20 camera on MIPI-CSI1, 3840x2160, 30 fps\n"
-USAGE+="\tos08a20_1080p30hdr      - single os08a20 camera on MIPI-CSI1, 1920x1080, 30 fps, HDR configuration\n"
-USAGE+="\tdual_os08a20_1080p30hdr - dual os08a20 cameras on MIPI-CSI1/2, 1920x1080, 30 fps, HDR configuration\n"
-USAGE+="\tos08a20_4khdr           - single os08a20 camera on MIPI-CSI1, 3840x2160, 15 fps, HDR configuration\n"
+USAGE=$USAGE"\tos08a20_1080p60         - single os08a20 camera on MIPI-CSI1, 1920x1080, 60 fps\n"
+USAGE=$USAGE"\tdual_os08a20_1080p60    - dual os08a20 cameras on MIPI-CSI1/2, 1920x1080, 60 fps\n"
+USAGE=$USAGE"\tos08a20_4k              - single os08a20 camera on MIPI-CSI1, 3840x2160, 30 fps\n"
+USAGE=$USAGE"\tos08a20_1080p30hdr      - single os08a20 camera on MIPI-CSI1, 1920x1080, 30 fps, HDR configuration\n"
+USAGE=$USAGE"\tdual_os08a20_1080p30hdr - dual os08a20 cameras on MIPI-CSI1/2, 1920x1080, 30 fps, HDR configuration\n"
+USAGE=$USAGE"\tos08a20_4khdr           - single os08a20 camera on MIPI-CSI1, 3840x2160, 15 fps, HDR configuration\n"
 
-USAGE+="\timx219_1080p60         - single imx219 camera on MIPI-CSI1, 1920x1080, 60 fps\n"
+USAGE=$USAGE"\timx219_1080p60         - single imx219 camera on MIPI-CSI1, 1920x1080, 60 fps\n"
 
 # parse command line arguments
 while [ "$1" != "" ]; do
@@ -181,21 +181,18 @@ load_module () {
 
 load_modules () {
 	# remove any previous instances of the modules
-	n=${#MODULES_TO_REMOVE[*]}
-	for (( i = n-1; i >= 0; i-- ))
+	for i in "${MODULES_TO_REMOVE}"
 	do
-		echo "Removing ${MODULES_TO_REMOVE[i]}..."
-		rmmod ${MODULES_TO_REMOVE[i]} &>/dev/null
-		#LSMOD_STATUS=`lsmod | grep "${MODULES[i]}"`
-		#echo "LSMOD_STATUS=$LSMOD_STATUS"
-		if lsmod | grep "${MODULES_TO_REMOVE[i]}" ; then
-			echo "Removing ${MODULES_TO_REMOVE[i]} failed!"
+		echo "Removing $i..."
+		rmmod $i &>/dev/null
+		if lsmod | grep "$i" ; then
+			echo "Removing $i failed!"
 			exit 1
 		fi
 	done
 
 	# and now clean load the modules
-	for i in "${MODULES[@]}"
+	for i in "${MODULES}"
 	do
 		echo "Loading module $i ..."
 		load_module $i
@@ -205,10 +202,10 @@ load_modules () {
 write_default_mode_files
 
 echo "Trying configuration \"$ISP_CONFIG\"..."
-MODULES_TO_REMOVE=("basler-camera-driver-vvcam" "os08a20" "ov2775" "imx219" "${MODULES[@]}")
+MODULES_TO_REMOVE="basler-camera-driver-vvcam os08a20 ov2775 imx219 ${MODULES}"
 case "$ISP_CONFIG" in
 		basler_4k )
-			MODULES=("basler-camera-driver-vvcam" "${MODULES[@]}")
+			MODULES="basler-camera-driver-vvcam ${MODULES}"
 			RUN_OPTION="CAMERA0"
 			CAM_NAME="basler-vvcam"
 			DRV_FILE="daA3840_30mc.drv"
@@ -217,7 +214,7 @@ case "$ISP_CONFIG" in
 			write_sensor_cfg_file "Sensor0_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
 			;;
 		basler_1080p60 )
-			MODULES=("basler-camera-driver-vvcam" "${MODULES[@]}")
+			MODULES="basler-camera-driver-vvcam ${MODULES}"
 			RUN_OPTION="CAMERA0"
 			CAM_NAME="basler-vvcam"
 			DRV_FILE="daA3840_30mc.drv"
@@ -226,7 +223,7 @@ case "$ISP_CONFIG" in
 			write_sensor_cfg_file "Sensor0_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
 			;;
 		dual_basler_1080p60 )
-			MODULES=("basler-camera-driver-vvcam" "${MODULES[@]}")
+			MODULES="basler-camera-driver-vvcam ${MODULES}"
 			RUN_OPTION="DUAL_CAMERA"
 			CAM_NAME="basler-vvcam"
 			DRV_FILE="daA3840_30mc.drv"
@@ -236,7 +233,7 @@ case "$ISP_CONFIG" in
 			write_sensor_cfg_file "Sensor1_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
 			;;
 		basler_4khdr )
-			MODULES=("basler-camera-driver-vvcam" "${MODULES[@]}")
+			MODULES="basler-camera-driver-vvcam ${MODULES}"
 			RUN_OPTION="CAMERA0"
 			CAM_NAME="basler-vvcam"
 			DRV_FILE="daA3840_30mc.drv"
@@ -245,7 +242,7 @@ case "$ISP_CONFIG" in
 			write_sensor_cfg_file "Sensor0_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
 			;;
 		basler_1080p60hdr )
-			MODULES=("basler-camera-driver-vvcam" "${MODULES[@]}")
+			MODULES="basler-camera-driver-vvcam ${MODULES}"
 			RUN_OPTION="CAMERA0"
 			CAM_NAME="basler-vvcam"
 			DRV_FILE="daA3840_30mc.drv"
@@ -254,7 +251,7 @@ case "$ISP_CONFIG" in
 			write_sensor_cfg_file "Sensor0_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
 			;;
 		dual_basler_1080p60hdr )
-			MODULES=("basler-camera-driver-vvcam" "${MODULES[@]}")
+			MODULES="basler-camera-driver-vvcam ${MODULES}"
 			RUN_OPTION="DUAL_CAMERA"
 			CAM_NAME="basler-vvcam"
 			DRV_FILE="daA3840_30mc.drv"
@@ -264,7 +261,7 @@ case "$ISP_CONFIG" in
 			write_sensor_cfg_file "Sensor1_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
 			;;
 		 os08a20_4k )
-			MODULES=("os08a20" "${MODULES[@]}")
+			MODULES="os08a20 ${MODULES}"
 			RUN_OPTION="CAMERA0"
 			CAM_NAME="os08a20"
 			DRV_FILE="os08a20.drv"
@@ -273,7 +270,7 @@ case "$ISP_CONFIG" in
 			write_sensor_cfg_file "Sensor0_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
 			;;
 		 os08a20_4khdr )
-			 MODULES=("os08a20" "${MODULES[@]}")
+			 MODULES="os08a20 ${MODULES}"
                          RUN_OPTION="DUAL_CAMERA"
                          CAM_NAME="os08a20"
                          DRV_FILE="os08a20.drv"
@@ -282,7 +279,7 @@ case "$ISP_CONFIG" in
                          write_sensor_cfg_file "Sensor0_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
 			 ;;
 		 os08a20_1080p60 )
-			 MODULES=("os08a20" "${MODULES[@]}")
+			 MODULES="os08a20 ${MODULES}"
                          RUN_OPTION="CAMERA0"
                          CAM_NAME="os08a20"
                          DRV_FILE="os08a20.drv"
@@ -291,7 +288,7 @@ case "$ISP_CONFIG" in
                          write_sensor_cfg_file "Sensor0_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
                          ;;
                  dual_os08a20_1080p60 )
-			 MODULES=("os08a20" "${MODULES[@]}")
+			 MODULES="os08a20 ${MODULES}"
                          RUN_OPTION="DUAL_CAMERA"
                          CAM_NAME="os08a20"
                          DRV_FILE="os08a20.drv"
@@ -301,7 +298,7 @@ case "$ISP_CONFIG" in
                          write_sensor_cfg_file "Sensor1_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
                          ;;
 		 os08a20_1080p30hdr )
-			 MODULES=("os08a20" "${MODULES[@]}")
+			 MODULES="os08a20 ${MODULES}"
                          RUN_OPTION="CAMERA0"
                          CAM_NAME="os08a20"
                          DRV_FILE="os08a20.drv"
@@ -310,7 +307,7 @@ case "$ISP_CONFIG" in
                          write_sensor_cfg_file "Sensor0_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
                          ;;
 		 dual_os08a20_1080p30hdr )
-			 MODULES=("os08a20" "${MODULES[@]}")
+			 MODULES="os08a20 ${MODULES}"
                          RUN_OPTION="DUAL_CAMERA"
                          CAM_NAME="os08a20"
                          DRV_FILE="os08a20.drv"
@@ -320,7 +317,7 @@ case "$ISP_CONFIG" in
                          write_sensor_cfg_file "Sensor1_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
                          ;;
                  imx219_1080p60 )
-                         MODULES=("imx219" "${MODULES[@]}")
+                         MODULES="imx219 ${MODULES}"
                          RUN_OPTION="CAMERA0"
                          CAM_NAME="imx219"
                          DRV_FILE="imx219.drv"
@@ -329,7 +326,7 @@ case "$ISP_CONFIG" in
                          write_sensor_cfg_file "Sensor0_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
                          ;;
                  dual_imx219_1080p60 )
-                         MODULES=("imx219" "${MODULES[@]}")
+                         MODULES="imx219 ${MODULES}"
                          RUN_OPTION="DUAL_CAMERA"
                          CAM_NAME="imx219"
                          DRV_FILE="imx219.drv"
-- 
2.34.1

